### PR TITLE
New recipients

### PR DESCRIPTION
**Pull request by ghost**
- adding eudat b2share basic api wrapping
- minor update to shipment management

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ Configuration is based on environment variables as shown in the table below. The
 `SHIPPER_MONGO_NAME` | `mongodb_db` | name of the MongoDB
 `SHIPPER_BOTTLE_HOST` | `bottle_host` | host for bottle, the WSGI micro web-framework used with shipper; default is `localhost`, to allows access from other local services running in containers, set this to `0.0.0.0`
 `SHIPPER_BOTTLE_PORT` | `bottle_port` | port for bottle
+`SHIPPER_REPO_EUDAT_HOST` | `repository_eudat_host` | host of Eudat b2share's API, defaults to _b2share Sandbox_
+`SHIPPER_REPO_EUDAT_TOKEN` | `repository_eudat_token` | API token for zenodo
```
Comment (nuest) [inline README.md:57]: Comment should be "API token for Eudat"

File: README.md [hunk 2]
```diff
@@ -64,7 +66,13 @@ Configuration is based on environment variables as shown in the table below. The
 
 ## Shipment recipients
 
+### Eudat b2share repository
+
+To ship ERC to [Eudat b2share](https://b2share.eudat.eu/) (or the [Eudat b2share Sandbox](https://trng-b2share.eudat.eu/)), you must create an account and log in.
+Then go to your account and get the personal access token.
+
+
```
Comment (nuest) [inline README.md]: Rest of file uses only one line break before and after headlines :-)

File: shipper.py [hunk 3]
```diff
@@ -126,13 +125,13 @@ def shipment_new():
                     'user': user_entitled,
                     'status': 'new',
                     'action': request.forms.get('action'),
-                    'md': new_md
+                    'md': new_md,
+                    'file_id': request.forms.get('file_id')
```
Comment (nuest) [inline shipper.py:129]: `file_id` is not documented yet in https://github.com/o2r-project/o2r-web-api/pull/41 - what does it do?
(same for `action` afaics)

File: shipper.py [hunk 4]
```diff
@@ -143,15 +142,28 @@ def shipment_new():
...
+                                data['status'] = 'deposited'
+                            elif data['recipient'] == 'eudat':
+                                data['deposition_id'] = eudat_create_depot(env_repository_eudat_host, env_repository_eudat_token)
+                                data['deposition_url'] = ''.join((env_repository_eudat_host.replace('api', 'records/'), data['deposition_id']))
+                                eudat_add_zip_to_depot(env_repository_eudat_host, data['deposition_id'], file_name, compendium_files, env_repository_eudat_token)
+                                # Add metadata that are in compendium in db:
+                                if 'metadata' in current_compendium:
+                                    if 'eudat' in current_compendium['metadata']:
+                                        md = current_compendium['metadata']['eudat']
+                                        zen_add_metadata(env_repository_eudat_host, data['deposition_id'], md,
```
Comment (nuest) [inline shipper.py]: Maybe rename method if it is not only for Zenodo anymore (I assume that is why the name starts with `zen_`).

File: shipper.py [hunk 5]
```diff
@@ -255,6 +283,71 @@ def session_user_entitled(cookie, min_lvl):
...
+            r = requests.get(''.join((base, '/records/', deposition_id, '/draft?access_token=', token)), headers=headers)
+            bucket_url = r.json()['links']['files']
+            if r.status_code == 200:
+                status_note(str(r.status_code) + ' using bucket <' + bucket_url + '>')
+            else:
+                status_note(r.status_code)
+            # upload file into bucket:
+            headers = {"Content-Type": "application/octet-stream"}
+            # create a filelike object in memory
+            filelike = BytesIO()
```
Comment (nuest) [inline shipper.py:316]: That is some nice code!

**Review by nuest — CHANGES_REQUESTED** [review]
Code looks fine, but when I make a POST request to create a new shipment, I get 

```json
{
  "error": "bad request: authentication cookie is missing"
}
```

though I'm using the same test request as before.

Can you provide a working example with curl?

File: README.md (hunk 1 above)
Comment (ghost) [reply README.md:57]: right.

File: README.md (hunk 2 above)
Comment (ghost) [reply README.md]: right.

File: shipper.py (hunk 3 above)
Comment (ghost) [reply shipper.py:129]: I'm a bit behind concerning the api docu.
`file_id` is only needed, when deleting a specific file from a deposition. if there is only one file in deposition, `file_id` can be `None`.
`action` is the parameter for CRUD actions in repository file management. This is required and probabliy the reason for the above stated "authentication" error. (providing too few parameters made it think, we miss the cookie that is the last parameter.

File: shipper.py (hunk 4 above)
Comment (ghost) [reply shipper.py]: this is a mistake. we must use the eudat equivalent. serving two or more repos made me replicate half of the functions. I need to find a way do this more efficiently...

File: shipper.py (hunk 5 above)
Comment (ghost) [reply shipper.py:316]: we already had a similar segment for zipping when it was only zenodo :)
I think it's a good way to stream the zip file right away during creation, however, it still needs to prove robustness. anyway this works while we move in the gigabytes...

**Comment by ghost** [comment]
fixed the minor issues. 
for documentation of new api parameters, see https://github.com/o2r-project/o2r-web-api/pull/41

**Pull request merged**